lib/bootloader_pvm.pm

```diff
@@ -170,6 +194,34 @@ sub boot_pvm {
     }
 }
 
+=head2 check_lpar_is_down
+
+ check_lpar_is_down($hmc_machine_name, $lpar_id);
+
+Check if an LPAR identified by C<$lpar_id> in the Power machine C<$hmc_machine_name> is down, by querying with the C<lssyscfg> command.
+Command will check 24 times in a loop while waiting 5 seconds between each run, and exit as soon as the LPAR is down. Check is performed
+by a needle match looking for the B<LPAR IS DOWN> text which is printed by the check script on success.
+
+=cut
+
+sub check_lpar_is_down {
+    my ($hmc_machine_name, $lpar_id) = @_;
+    enter_cmd("for ((i=0\; i<24\; i++)); do lssyscfg -m $hmc_machine_name -r lpar --filter \"\"lpar_ids=$lpar_id\"\" -F state | grep -q 'Not Activated' && echo 'LPAR IS DOWN' && break || echo 'Waiting for lpar $lpar_id to shutdown' && sleep 5 ; done ");
```

Comment:
why the for is there and not in perl?

Reply:
It was there already. I just moved it to a function as the changes I'm introducing require to call this piece of code twice and didn't want to repeat myself.